.github/workflows/publish-toolkit.yml

```diff
@@ -0,0 +1,73 @@
+name: Publish Toolkit
+
+on:
+  workflow_dispatch:
+    inputs:
+      toolkit:
+        description: 'The directory of the toolkit to publish'
+        required: true
+
+jobs:
+  release-and-publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+
+    steps:
+    - name: Set toolkit input
+      id: set-toolkit
+      run: |
+        if [ "${{ github.event_name }}" = "push" ]; then
+          # Extract toolkit name from changed files
+          TOOLKIT=this_is_for_testing
+          echo "toolkit=$TOOLKIT" >> $GITHUB_OUTPUT
+        else
+          echo "toolkit=${{ inputs.toolkit }}" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        cache: 'pip'
+
+    - name: Install Python Dependencies
+      run: pip install ./arcade
+
+    - name: Test Toolkit
+      id: Test_Toolkit
+      working-directory: toolkits/${{ steps.set-toolkit.outputs.toolkit }}
+      run: |
+        make install
+        make test
```

Comment:
should we add `make check` as well?

Reply:
yeah probably

Reply:
or maybe that should be done before merging to main?

Reply:
we should probably have another workflow that tests and checks them all on commit